src/graph/backend/dnnl/kernels/sdp_primitive.cpp

```diff
@@ -45,6 +45,17 @@ status_t sdp_primitive_kernel_t<quantized>::compile_impl(
         const dnnl_partition_impl_t *part, const engine_t *g_engine,
         const std::vector<logical_tensor_t> &inputs,
         const std::vector<logical_tensor_t> &outputs) {
+// sdp_primitive_kernel_t only supports Intel GPU.
+#ifdef DNNL_WITH_SYCL
+#if DNNL_GPU_VENDOR != DNNL_VENDOR_INTEL
+    UNUSED(part);
+    UNUSED(g_engine);
+    UNUSED(inputs);
+    UNUSED(outputs);
+    return status::unimplemented;
+#endif
+#endif
```

Comment:
Variables will still be formally used behind the return.
```suggestion
#if defined(DNNL_WITH_SYCL) && DNNL_GPU_VENDOR != DNNL_VENDOR_INTEL
    return status::unimplemented;
#endif
```

If somehow it doesn't, this may:
```suggestion
#if defined(DNNL_WITH_SYCL)
    if (DNNL_GPU_VENDOR != DNNL_VENDOR_INTEL) return status::unimplemented;
#endif
```